### PR TITLE
Applying metadata and adding presenter video

### PR DESCRIPTION
**Pull request by ferishili**
The catalog sample has been attached
[mdl_block_opencast_catalog.zip](https://github.com/unirz-tu-ilmenau/moodle-block_opencast/files/3418049/mdl_block_opencast_catalog.zip)

It requires a critical update on tool_opencast 
https://github.com/unirz-tu-ilmenau/moodle-tool_opencast/pull/17/commits/83ef0f6da9b757cd2b003b66a88dc15bfb9cd975

### CODE REVIEW

File: classes/local/apibridge.php [hunk 1]
```diff
@@ -1404,4 +1421,65 @@ public static function set_testing($testing) {
...
+     * @param bool the capability of updating!
+     */
+    public function can_update_event_metadata($video, $courseid) {
+
+        if (isset($video->processing_state) &&
+            ($video->processing_state !== 'RUNNING' && $video->processing_state !== 'PAUSED')) {
+
+            $context = \context_course::instance($courseid);
+            
+            //TODO: Add capabiliy to the block/opencast
```
Comment (rtschu) [inline classes/local/apibridge.php]: Todo should be resolved before merging

File: classes/local/apibridge.php [hunk 2]
```diff
@@ -1404,4 +1421,65 @@ public static function set_testing($testing) {
...
+            return has_capability('block/opencast:addvideo', $context);
+            // return true;
```
Comment (rtschu) [inline classes/local/apibridge.php]: there shouldn't be commented out code

File: classes/local/updatemetadata_form.php [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            "nor" => ucwords(strtolower("NORWEGIAN")),
+            "rus" => ucwords(strtolower("RUSSIAN")),
+            "jpx" => ucwords(strtolower("JAPANESE")),
+            "nld" => ucwords(strtolower("DUTCH")),
+            "tur" => ucwords(strtolower("TURKISH")),
+            "hin" => ucwords(strtolower("HINDI")),
+            "swa" => ucwords(strtolower("SWEDISH")),
+            "eng" => ucwords(strtolower("ENGLISH")),
+            "deu" => ucwords(strtolower("GERMAN"))
+        ];
```
Comment (rtschu) [inline classes/local/updatemetadata_form.php]: Why use ucwords and strtolower? Barely improves readability and adds functioncalls

File: classes/local/updatemetadata_form.php [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                if ($date_time) {
+                    $field->value = $date_time;
+                }
+            }
+            if ($field->id == 'description') {$type = 'textarea';}
+            if ($field->id == 'language') {
+                $type = 'select';
+                /* $param = array_map(function($val){
+                    return ucwords(strtolower($val));
+                }, $languages); */
```
Comment (rtschu) [inline classes/local/updatemetadata_form.php]: there shouldn't be commented out code

**Comment by ferishili** [comment]
This pull request has been updated and it is not the latest intended version!

File: adminsettings.php [hunk 5]
```diff
@@ -70,6 +78,16 @@
...
         echo $OUTPUT->footer();
         exit();
+    } else if (!empty($delcatalog)) {
+        // Deletion has to be confirmed.
+        // Print a confirmation message.
+        echo $OUTPUT->header();
+        echo $OUTPUT->heading(get_string('settings', 'block_opencast'));
+        echo $OUTPUT->confirm(get_string("delete_confirm", 'block_opencast'),
+            "adminsettings.php?cd=$delcatalog&c=$delcatalog",
+            'adminsettings.php');
```
[the remark this answers is not in the document]
Comment (ferishili) [reply adminsettings.php]: In order to have consistency, I followed the previous coding styles wherever I could!

File: adminsettings.php [hunk 6]
```diff
@@ -60,6 +61,13 @@
         exit();
     }
 
+    if (!empty($delcatalog) && !empty($confirm)) {
```
[the remark this answers is not in the document]
Comment (ferishili) [reply adminsettings.php]: In order to have consistency, I followed the previous coding styles wherever I could!

**Comment by tobiasreischmann** [comment]
All remarks in here have been cared for. Thanks @ferishili

**Pull request closed without merging**